content/content/oop_macro.md

```diff
@@ -50,9 +50,9 @@ macro class*(head, body: untyped): untyped =
   var typeName, baseName: NimNode
 
   # flag if object should be exported
-  var exported: bool
+  var isExported: bool
 
-  if head.kind == nnkInfix and head[0].ident == !"of":
+  if head.kind == nnkInfix and $head[0] == "of":
```

Comment:
Shouldn't `eqIdent` be used instead?

Reply:
That's right, thank you! I'll be sure to fix that after merging.

Reply:
No its fine as it is.

Reply:
its just "*" and "of", eqIdent is not needed.

Reply:
Still, [the docs](https://nim-lang.org/docs/macros.html#eqIdent,NimNode,string) say it is the preferred way to check, and since this is a tutorial it should be correct

Reply:
i don not agree with you. Everyone is entitled to his opinion and I respect yours. But yet since I have been maintaining this macro for some time, I kindly ask @flaviut not to change it.

Reply:
oh, i just remembered eqIdent would allow stuff like `o___F`.  i just thought there might have been a more idiomatic way to check if the node is "of" and eqIdent popped to mind but i forgot it ignores style. I guess thats why you were holding off

Reply:
Yes this bothers me too!

Reply:
this conversation led me to test out the following code:

```
pR_oc abc(): int =
  return 1
```

Turns out that it's valid code. Since one of the nice things about macros in Nim is that they're indistinguishable from regular language constructs, I'd like to make sure that `o___F` *is* allowed.  

To be honest, I'd prefer Nim enforce a particular style, kind of like Go. But that's not the direction that the language went in, and I believe that consistency with everything else should override my personal preferences.

@notTito I'd like to hear your thoughts on this.

Reply:
eqIdent should be used.